grpc-bridge-runtime/src/main/java/org/jboss/resteasy/grpc/runtime/servlet/HttpServletResponseImpl.java

```diff
@@ -0,0 +1,440 @@
+      }
+      // if no explicit encoding is specified, this method is supposed to return ISO-8859-1 as per the
+      // expectation of this API
+      return StandardCharsets.ISO_8859_1.name();
+   }
+
+   @Override
+   public String getContentType() {
+      if (contentType != null) {
+         if (charsetSet) {
+            return contentType + ";charset=" + getCharacterEncoding();
+         } else {
+            return contentType;
+         }
+      }
+      return null;
+   }
+
+   @Override
+   public ServletOutputStream getOutputStream() throws IOException {
+      if (responseState == ResponseState.WRITER) {
+         throw Messages.MESSAGES.getWriterAlreadyCalled();
+      }
+      responseState = ResponseState.STREAM;
+      return msos;
+   }
+
+   @Override
+   public PrintWriter getWriter() throws IOException {
+      if (writer == null) {
+         if (!charsetSet) {
+            //servet 5.5
+            setCharacterEncoding(getCharacterEncoding());
+         }
+         if (responseState == ResponseState.STREAM) {
+            throw Messages.MESSAGES.getOutputStreamAlreadyCalled();
+         }
+         responseState = ResponseState.WRITER;
+         writer = new PrintWriter(msos);
+      }
+      return writer;
+   }
+
+   @Override
+   public void setCharacterEncoding(final String charset) {
+      charsetSet = charset != null;
+      this.charset = charset;
+      if (contentType != null) {
+         headers.put(HttpHeaders.CONTENT_TYPE, toList(getContentType()));
+      }
+   }
+
+   @Override
+   public void setContentLength(int len) {
+   }
+
+   @Override
+   public void setContentLengthLong(long len) {
+   }
+
+   @Override
+   public void setContentType(String type) {
+      if (type == null) {
+         return;
+      }
+      ContentTypeInfo ct = parseContentType(type);
+      contentType = ct.getContentType();
+      boolean useCharset = false;
+      if(ct.getCharset() != null && writer == null && !isCommitted()) {
+         charset = ct.getCharset();
+         charsetSet = true;
+         useCharset = true;
+      }
+      if(useCharset || !charsetSet) {
+         headers.put(HttpHeaders.CONTENT_TYPE, toList(ct.getHeader()));
+      } else if(ct.getCharset() == null) {
+         headers.put(HttpHeaders.CONTENT_TYPE, toList(ct.getHeader() + "; charset=" + charset));
+      }else {
+         headers.put(HttpHeaders.CONTENT_TYPE, toList(ct.getContentType() + "; charset=" + charset));
+      }
+   }
+
+   @Override
+   public void setBufferSize(int size) {
+      // no-op
+   }
+
+   @Override
+   public int getBufferSize() {
+      return 0;
+   }
+
+   @Override
+   public void flushBuffer() throws IOException {
+      // no-op
+   }
+
+   @Override
+   public void resetBuffer() {
+      // no-op
+   }
+
+   @Override
+   public boolean isCommitted() {
+      // TODO Auto-generated method stub
+      return false;
+   }
+
+   @Override
+   public void reset() {
+      msos.getDelegate().reset();
```

Comment:
Can this just be `msos.reset()`? It's weird this would have to know it's a delegate.

Reply:
Done